### ISSUE TITLE
E0601 `used-before-assignment` false positive with `TYPE_CHECKING`

### ISSUE THREAD

**Issue by yukihiko-shinoda** [PyCQA/pylint#8376]
### Bug description

a.py:

```python
"""Test."""
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from pathlib import Path


def test(path: Path) -> None:
    """Test."""
    path.resolve()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
test.py:9:15: E0601: Using variable 'Path' before assignment (used-before-assignment)
```

### Expected behavior

No warnings.

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.2 (main, Feb  9 2023, 05:12:33) [GCC 10.2.1 20210110]
```

### OS / Environment

```console
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

### Additional dependencies

```console
$pipenv graph
pylint==2.16.2
  - astroid [required: >=2.14.2,<=2.16.0-dev0, installed: 2.14.2]
    - lazy-object-proxy [required: >=1.4.0, installed: 1.9.0]
    - wrapt [required: >=1.14,<2, installed: 1.15.0]
  - dill [required: >=0.3.6, installed: 0.3.6]
  - isort [required: >=4.2.5,<6, installed: 5.12.0]
  - mccabe [required: >=0.6,<0.8, installed: 0.6.1]
  - platformdirs [required: >=2.2.0, installed: 3.0.0]
  - tomlkit [required: >=0.10.1, installed: 0.11.6]
```

**Comment by yukihiko-shinoda** [comment]
Very sorry, this was my fault...

```diff
"""Test."""
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from pathlib import Path


- def test(path: Path) -> None:
+ def test(path: "Path") -> None:
    """Test."""
    path.resolve()
```

**Comment by Pierre-Sassoulas** [comment]
Note that you could also do:
```
"""Test."""

from __future__ import annotations

from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from pathlib import Path


def test(path: Path) -> None:
    """Test."""
    path.resolve()
```
